### ISSUE TITLE
 trying to upload an IPA file to App Store Connect

### ISSUE THREAD

**Issue by eslamoomohamed** [fastlane/fastlane#21115]
### New Issue Checklist

- [x] Updated fastlane to the latest version
- [x] I read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] I read [docs.fastlane.tools](https://docs.fastlane.tools)
- [x] I searched for [existing GitHub issues](https://github.com/fastlane/fastlane/issues)

### Issue Description

I'm trying to upload an IPA file to App Store Connect, but for some reason it keeps trying to upload the .itmsp folder instead of the IPA file itself. I've already specified the path to the IPA file in fastfile, so I'm not sure what else could be causing this issue.
it's already uploading to firebase app distribution, but for App Store no.
I'm using Xcode 14.2 and fastlane 2.212.1
I'd appreciate any help or advice that you can give me. Thank you in advance!

##### Command executed

  upload_to_testflight(
    ipa: ipa_path,
    api_key: api_key,
    team_id: "********",
    groups: "********",
    skip_waiting_for_build_processing: true,
    app_identifier: "*********",
    distribute_external: true,
    notify_external_testers: true,
    changelog: changelog_from_git_commits( commits_count: "12")
  )

##### Complete output when running fastlane, including the stack trace and command used

[14:45:38]: [altool] 2023-03-07 14:45:38.807 *** Error: Could not determine the package’s bundle ID. The package is missing an Info.plist or the CFBundlePackageType is not ‘APPL’ or ‘FMWK’. Unable to validate your application. (-21017)
[3100](https://github.com/MaticServices/matic-rosie-ios/actions/runs/4353654947/jobs/7608153222#step:2:3101)

[3101](https://github.com/MaticServices/matic-rosie-ios/actions/runs/4353654947/jobs/7608153222#step:2:3102)
[14:45:38]: [altool]  {
[3102](https://github.com/MaticServices/matic-rosie-ios/actions/runs/4353654947/jobs/7608153222#step:2:3103)

[3103](https://github.com/MaticServices/matic-rosie-ios/actions/runs/4353654947/jobs/7608153222#step:2:3104)
[14:45:38]: [altool]     NSLocalizedDescription = "Could not determine the package\U2019s bundle ID. The package is missing an Info.plist or the CFBundlePackageType is not \U2018APPL\U2019 or \U2018FMWK\U2019.";
[3104](https://github.com/MaticServices/matic-rosie-ios/actions/runs/4353654947/jobs/7608153222#step:2:3105)

[3105](https://github.com/MaticServices/matic-rosie-ios/actions/runs/4353654947/jobs/7608153222#step:2:3106)
[14:45:38]: [altool]     NSLocalizedFailureReason = "Unable to validate your application.";
[3106](https://github.com/MaticServices/matic-rosie-ios/actions/runs/4353654947/jobs/7608153222#step:2:3107)

[3107](https://github.com/MaticServices/matic-rosie-ios/actions/runs/4353654947/jobs/7608153222#step:2:3108)
[14:45:38]: [altool] }
[3108](https://github.com/MaticServices/matic-rosie-ios/actions/runs/4353654947/jobs/7608153222#step:2:3109)

[3109](https://github.com/MaticServices/matic-rosie-ios/actions/runs/4353654947/jobs/7608153222#step:2:3110)
[14:45:38]: Application Loader output above ^
[3110](https://github.com/MaticServices/matic-rosie-ios/actions/runs/4353654947/jobs/7608153222#step:2:3111)
[14:45:38]: Error uploading '/var/folders/hd/6j1th0bd12j_ftvqvd69ml180000gn/T/d20230307-76936-zll04u/984979642-c211db82-9d1e-418a-bb80-84bae6dab65f.itmsp'.
[3111](https://github.com/MaticServices/matic-rosie-ios/actions/runs/4353654947/jobs/7608153222#step:2:3112)

[3112](https://github.com/MaticServices/matic-rosie-ios/actions/runs/4353654947/jobs/7608153222#step:2:3113)
[14:45:38]: Could not determine the package’s bundle ID. The package is missing an Info.plist or the CFBundlePackageType is not ‘APPL’ or ‘FMWK’. Unable to validate your application. (-21017)
[3113](https://github.com/MaticServices/matic-rosie-ios/actions/runs/4353654947/jobs/7608153222#step:2:3114)

[3114](https://github.com/MaticServices/matic-rosie-ios/actions/runs/4353654947/jobs/7608153222#step:2:3115)
[14:45:38]: The call to the altool completed with a non-zero exit status: 1. This indicates a failure.
[3115](https://github.com/MaticServices/matic-rosie-ios/actions/runs/4353654947/jobs/7608153222#step:2:3116)
[14:45:38]: Could not download/upload from App Store Connect!
<details>
  <pre> [REPLACE THIS WITH YOUR INFORMATION] </pre>
</details>

### fastlane gems

| Gem      | Version | Update-Status |
| -------- | ------- | ------------- |
| fastlane | 2.212.1 | ✅ Up-To-Date  |
<details>
  <pre> [REPLACE THIS WITH YOUR INFORMATION] </pre>
</details>

**Comment by muhammadqazi** [comment]
@eslamoomohamed try this guide 
https://medium.com/@muhammedqazi/mastering-fastlane-the-ultimate-guide-to-effortless-android-and-ios-deployments-with-github-e0878c3f9529

**Comment by tomwanzek** [comment]
@eslamoomohamed Not sure, if it is the same scenario for you, but I ran into the same issue after upgrading some iOS Build dependencies for our CircleCI job. We moved from  a slightly older fastlane + XCode 13.x build & deploy pipeline to:
* fastlane 2.212.1, XCode 14.2, cocoapods 1.12.0 (cocoapods 1.12.0 was critical to remove issues with the actual build/signing)

With those dependencies we ran into the issue you described.

The good news is, we got through it by removing a temporary tweak he had to introduce 5 months or so ago.
Back then we had to add

```ruby
ENV['ITMSTRANSPORTER_FORCE_ITMS_PACKAGE_UPLOAD'] = "true"
```

As per https://github.com/fastlane/fastlane/issues/20813 this seems be an issue with the newer versions.

Once we removed it, it unblocked this issue.

The downside is we immediately ran into the next one, which appears to be essentially https://github.com/fastlane/fastlane/issues/20799

We're still investigating how to unblock that one.
